src/main/java/it/gov/pagopa/pu/registry/service/pagopa/PagoPaRegistryService.java

```diff
@@ -27,4 +32,36 @@ public void consumePaymentEvent(RegistryEventPagoPaDTO event) {
     pagopaRegistryRepository.saveAll(registry);
   }
 
+  public PagoPaRegistryDTO getPagoPaRegistry(String registryId) {
+    var opt = pagopaRegistryRepository.findById(registryId);
+
+    if (opt.isEmpty()) {
+      log.error("No registry found with id: {}", registryId);
+      throw new ApplicationRestException("No registry found with id: " + registryId, HttpStatus.NOT_FOUND);
```

Comment:
throw an exception more meaningfull.. like a ResourceNotFoundException...

Reply:
never add presentation layer info on business logic exception: the configuration of http status to be returned directly on the exceptions will never be allowed.

we will use the controllerAdvice to intercept specific exception and translate them on the needed http status.